app/views/shared/_header.html.haml

```diff
@@ -1,22 +1,22 @@
 %section#top-header.row
   %h1.col-md-2
-    = link_to "MyFLiX"
-  %ul.col-md-4.clearfix
-    %li= link_to "Videos"
-    %li= link_to "My Queue"
-    %li= link_to "People"
-  %form.col-md-5.navbar-form(action="")
-    .form-group
-      %input.form-control(type="text" placeholder="Search for videos here")
-    %button.btn.btn-default(type="submit") Search
-  #user_links.pull-right
-    %ul
-      %li.dropdown
-        %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
-          Welcome, Philip J. Fry
-          %b.caret
-        %ul.dropdown-menu(role="menu" aria-labelledby="dlabel")
-          %li
-            %a(href="#") Account
-            %a(href="#") Plan and Billing
-            %a(href="#") Sign Out
+    = link_to "MyFLiX", root_path
+  -if logged_in?
+    %ul.col-md-4.clearfix
+      %li= link_to "Videos"
+      %li= link_to "My Queue"
+      %li= link_to "People"
+    =form_tag(search_videos_path, class: "col-md-5 navbar-form form-group" ) do
+      =text_field_tag 'title', nil, class: 'form-control', placeholder: "Search for videos here"
+      =submit_tag "search", class: 'btn btn-default'
+    #user_links.pull-right
+      %ul
+        %li.dropdown
+          %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
+            ="Welcome, #{current_user.name}" 
```

Comment:
I tried to make this adjustment but I'm getting a  'dynamic constant assignment' error :|, i tried googling this and looked at the HAML documentation. Cannot seem to figure out why your syntax doesn't work for me.